massa-consensus-worker/src/state/mod.rs

```diff
@@ -331,19 +337,20 @@ impl ConsensusState {
         // add all the active blocks that are after the effective_latest_finals of their thread
         self.add_active_blocks_after(&mut kept_blocks, &effective_latest_finals, end_slot);
 
-        // do the following 3 times
-        for _ in 0..3 {
+        // do the following 2 times
+        for _ in 0..2 {
             // extend kept_blocks with the parents of the current kept_blocks
-            let kept_blocks_clone = kept_blocks.clone();
-            for id in kept_blocks_clone.iter() {
+            let mut cumulated_parents: Vec<BlockId> = Vec::new();
```

Comment:
This might get very big because of heavy redundancy.
We should use a prehashset

Reply:
Ah yes true mb

Reply:
Done